### PR TITLE
Fix radio off-centered dot and background colors

### PR DESCRIPTION
**Pull request by homing1**
Task: task-[580494](https://dev.azure.com/ceapex/Engineering/_workitems/edit/580494/)

Link: preview-[394](https://design.docs.microsoft.com/pulls/394/components/radio.html)

This PR addresses three issues for radios:
- Sometimes the radio dot is off-centered when checked, depending on screen size and zoom level. This approach fills the entire radio, then uses box shadow to add the white space. 
- The body background color wasn't filling the entire unchecked radio when paired with a container with `background-color-body-medium`. The entire radio should be white when unchecked.
- The color on hover on `is-checked` was not matching the checked color. The hover color should be the same.

## Testing

1. Visit https://design.docs.microsoft.com/pulls/394/components/radio.html .
2. Try different screen sizes and zoom levels. The dot should stay in the center. 
3. Add `background-color-body-medium` on a radio's parent container. The entire circle should have a white background when unchecked.
4. Add `is-checked` to an unchecked radio. There should be no color change on hover.
5. Test with OS high contrast or browser forced colors

### CODE REVIEW

**Review by wibjorn — APPROVED** [review]

File: css/src/components/form/radio.scss [hunk 1]
```diff
@@ -10,62 +9,35 @@ $radio-duration: $input-transition-duration !default;
...
-	margin: auto;
-	border-radius: $border-radius-rounded;
-	background-color: $body-background;
-	content: '';
-	overflow: hidden;
-}
-
 @mixin radio-dot-checked {
 	background: $radio-dot-checked-color;
+	box-shadow: inset 0 0 0 0.25em $body-background;
```
Comment (jdanyow) [inline css/src/components/form/radio.scss:14]: The dots disappear in forced colors mode

**Review by jdanyow — CHANGES_REQUESTED** [review]

File: css/src/components/form/radio.scss (hunk 1 above)
Comment (jdanyow) [reply css/src/components/form/radio.scss:14]: might be able to use this technique https://github.com/microsoft/atlas-design/blob/main/css/src/components/toggle.scss#L25

Comment (homing1) [reply css/src/components/form/radio.scss:14]: yup looking at that right now

Comment (homing1) [reply css/src/components/form/radio.scss:26]: Forced colors sets box shadow to none, so I think we can use the previous strategy of `:before` for forced colors only. This means the dot can still look weird in forced colors mode though. 

image

Comment (wibjorn) [reply css/src/components/form/radio.scss:26]: That will work, just verifying `forced-color-adjust: none;` was used inside the mixin body during testing though before we go that route.

```scss
@include forced-colors {
  forced-color-adjust: none;
  [... etc]
}
```

Comment (homing1) [reply css/src/components/form/radio.scss:26]: Tested using both with and without `forced-color-adjust: none;` using all four OS high contrast settings.

File: css/src/components/form/radio.scss [hunk 2]
```diff
@@ -42,30 +40,26 @@ $radio-spacing: 0.5em !default;
 		height: $radio-circle-size;
 		border: $radio-border-width solid $radio-border-color;
 		border-radius: $border-radius-rounded;
+		background-color: $body-background;
 		color: $radio-dot-color;
 		appearance: none;
+		content: '';
```
Comment (ollips) [inline css/src/components/form/radio.scss:46]: nit: can you add here `cursor: pointer;`? For some reason, I get the default arrow, would be nice to see the pointer :)

Comment (homing1) [reply css/src/components/form/radio.scss]: Added!

**Review by ollips — APPROVED** [review]

**Review by jdanyow — APPROVED** [review]

**Pull request merged**